src/runtime/runtime.ts

```diff
@@ -960,27 +960,46 @@ class JoinNode implements Node {
       return results;
     }
 
+
     let {proposer} = bestProposal;
     // We have to copy here because we need to keep a reference to this even if later
     // rounds might overwrite the proposal
     moveArray(bestProposal.forRegisters, forRegisters);
-    let resolved = proposer.resolveProposal(index, prefix, bestProposal, transaction, round, proposedResults);
-    resultLoop: for(let result of resolved) {
-      let ix = 0;
-      for(let register of forRegisters) {
-        prefix[register.offset] = result[ix];
-        ix++;
-      }
-      for(let constraint of constraints) {
-        if(constraint === proposer) continue;
-        if(!constraint.accept(index, prefix, transaction, round, forRegisters)) {
-          continue resultLoop;
+    let resolved:any[] = proposer.resolveProposal(index, prefix, bestProposal, transaction, round, proposedResults);
```

Comment:
The resolution can be either an `ID[]` or an `ID[][]`, but the typechecker didn't feel like cooperating based on the if statement on the next line